### ISSUE TITLE
Allow materializer targets to specify inputs

### ISSUE THREAD

**Issue by elijahbenizzy** [DAGWorks-Inc/hamilton#713]
**Is your feature request related to a problem? Please describe.**
I want to be able to use materializers to create simple ETLs of just moving data from one place to another.

**Describe the solution you'd like**

A little odd but a good use-case. Specifically, I want something like this:

```python
dr = driver.Builder().build() # empty driver
dr.materialize(
    to.csv(
        id="save_some_input_to_csv",
        dependencies=["some_input"]),
    inputs={
        "some_input" : SOME_DATAFRAME
    }
),
```

This should form a two-node DAG.

**Describe alternatives you've considered**
Just writing a function to do this, or doing it outside of Hamilton.

**Comment by skrawcz** [comment]
Doesn't this already work?

**Comment by skrawcz** [comment]
Yeah it's because it's empty it doesn't work. With anything defined it seems to work.
![Screen Shot 2024-02-23 at 5 54 29 PM](https://github.com/DAGWorks-Inc/hamilton/assets/2328071/d5211d37-ccd3-45d8-8545-c20050dbf571)

**Comment by elijahbenizzy** [comment]
> Yeah it's because it's empty it doesn't work. It needs a node with that input defined right now to work.
> 
> ![Screen Shot 2024-02-23 at 5 54 29 PM](https://github.com/DAGWorks-Inc/hamilton/assets/2328071/d5211d37-ccd3-45d8-8545-c20050dbf571)
> 
> 

Yep that's why I opened up this issue, cause I think it should work! It's a nice way to use materializers and track them. To avoid pains/encourage early failures we can add a validate_inputs param and default it to true.

**Comment by skrawcz** [comment]
closing this since it's partially addressed with #911 .
